apps/mobile-wallet/src/features/ecosystem/DAppWebViewScreen.tsx

```diff
@@ -1,48 +1,157 @@
+  const [canGoBack, setCanGoBack] = useState(false)
+  const [canGoForward, setCanGoForward] = useState(false)
 
   useDetectWCUrlInClipboardAndPair()
 
   if (!dAppUrl) return null
 
+  const handleGoBack = () => webViewRef.current?.goBack()
+
+  const handleGoForward = () => webViewRef.current?.goForward()
+
+  const handleReload = () => webViewRef.current?.reload()
+
+  const handleNavigationStateChange = (navState: WebViewNavigation) => {
+    setCanGoBack(navState.canGoBack)
+    setCanGoForward(navState.canGoForward)
+  }
+
   return (
-    <Screen
-      titleAlwaysVisible
-      headerOptions={{
-        type: 'stack',
-        headerTitle: dAppUrl,
-        headerRight: () => <AddToFavoritesButton dAppName={dAppName} />
-      }}
-      {...props}
-    >
-      <WebViewStyled source={{ uri: dAppUrl }} allowsBackForwardNavigationGestures pullToRefreshEnabled />
+    <Screen {...props}>
+      <BrowserHeader
+        dAppName={dAppName}
+        currentUrl={currentUrl}
+        onGoBack={handleGoBack}
+        onGoForward={handleGoForward}
+        onReload={handleReload}
+        canGoBack={canGoBack}
+        canGoForward={canGoForward}
+      />
+      <WebViewStyled
+        ref={webViewRef}
+        source={{ uri: dAppUrl }}
+        allowsBackForwardNavigationGestures
+        pullToRefreshEnabled
+        onLoad={(e) => setCurrentUrl(e.nativeEvent.url)}
+        onNavigationStateChange={handleNavigationStateChange}
+      />
     </Screen>
   )
 }
 
 export default DAppWebViewScreen
 
+interface BrowserHeaderProps {
+  dAppName: string
+  currentUrl: string
+  onGoBack: () => void
+  onGoForward: () => void
+  onReload: () => void
+  canGoBack: boolean
+  canGoForward: boolean
+}
+
+const BrowserHeader = ({
+  dAppName,
+  currentUrl,
+  onGoBack,
+  onGoForward,
+  onReload,
+  canGoBack,
+  canGoForward
+}: BrowserHeaderProps) => {
+  const navigation = useNavigation()
+  const insets = useSafeAreaInsets()
+
+  return (
+    <BrowserHeaderStyled style={{ paddingTop: insets.top }}>
+      <Button onPress={navigation.goBack} iconProps={{ name: 'arrow-left' }} squared compact />
+      <Url truncate color="secondary">
+        {currentUrl}
+      </Url>
+      <ButtonsList>
+        <AddToFavoritesButton dAppName={dAppName} />
+
+        {Platform.OS === 'android' && (
+          <>
+            <Button
+              onPress={onGoBack}
+              iconProps={{ name: 'arrow-left' }}
+              squared
+              type="transparent"
+              compact
+              disabled={!canGoBack}
+            />
+            <Button
+              onPress={onGoForward}
+              iconProps={{ name: 'arrow-right' }}
+              squared
+              type="transparent"
+              compact
+              disabled={!canGoForward}
+            />
+            <Button onPress={onReload} iconProps={{ name: 'refresh-cw' }} squared type="transparent" compact />
+          </>
+        )}
```

Comment:
I see that this is available for android only. Why is that?

Reply:
Ok, pull to refresh seems to be available only on iOS.

I think we could relocate the back / forward and refresh action in a bottom bar, available on Bothe platforms :)

Reply:
> Ok, pull to refresh seems to be available only on iOS.

Exactly, on iOS you can use gestures and pull to refresh works. So I thought why clutter the UI?

> I think we could relocate the back / forward and refresh action in a bottom bar, available on both platforms

That would reduce the visual space available for the dApp. I think it looks good keeping everything on top! If you strongly disagree, I can implement your suggestion. But I'd like to avoid having a full row used by some small buttons. Up to you

Reply:
Ideal would be smth like iOS, where the bottom bar is hidden when scrolling down. But may be hard / impossible to implement.

Having all the actions at the top makes it a bit crowded, hard to interact with on small screens. Buttons are super smol and only a small portion of the URL is shown.

Reply:
> Exactly, on iOS you can use gestures and pull to refresh works. So I thought why clutter the UI?

Gestures don't work for me. Siping left goes back to the previous app screen.

Reply:
OK, I'll move them to bottom and update the PR

Reply:
Hm, gestures work for me

Reply:
@mvaivre buttons moved to footer!